internal/clients/sources/sources_impl.go

```diff
@@ -2,14 +2,75 @@ package sources
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
 
 	"github.com/RHEnVision/provisioning-backend/internal/config"
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
 )
 
+var appTypeId string
+var lazyLoad sync.Mutex
+
+type AppType struct {
+	Id          string `json:"id"`
+	Name        string `json:"name"`
+	DisplayName string `json:"display_name"`
+}
+
+type dataElement struct {
+	Data []AppType `json:"data"`
+}
+
+func getAppTypeId() string {
+	lazyLoad.Lock()
+	defer lazyLoad.Unlock()
+	return appTypeId
+}
```

Comment:
Here I have the biggest concern for the future - the caller can accidentally call this function before getSourcesClient and it is not very obvious at all that this must be called after, otherwise locking will not even help and the problem persists - a goroutine/thread can cache an empty value.

So I suggest the solution - remove locking from this function and simply return error or just panic if this is called with the uninitialized value. This will prevent from accidental calls in the future.

Alternatively, you could use https://pkg.go.dev/sync/atomic to ensure the value is never cached, but you'd need to create a string wrapper (Generics hello!). I am fine with just panicking.

Reply:
Ok, panicing sounds good here I believe ^_^